internal/operands/common-templates/resource.go

```diff
@@ -0,0 +1,136 @@
+)
+
+const (
+	goldenImagesNSname = "kubevirt-os-images"
+	version            = "v0.12.2"
+)
+
+func clusterRoleName(role string) string {
+	return "os-images.kubevirt.io" + ":" + role
+}
+
+func loadTemplates(filename string) ([]templatev1.Template, error) {
+	var bundle []templatev1.Template
+	file, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+	for {
+		template := templatev1.Template{}
+		err = decoder.Decode(&template)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		if template.Name != "" {
+			bundle = append(bundle, template)
+		}
+	}
+	return bundle, nil
+}
+
+func newGoldenImagesNS(namespace string) *core.Namespace {
+	return &core.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+}
+
+func newViewRole(namespace string) *rbac.ClusterRole {
+	return &rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterRoleName("view"),
+			Namespace: namespace,
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"persistentvolumeclaims", "persistentvolumeclaims/status"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"cdi.kubevirt.io"},
+				Resources: []string{"datavolumes/source"},
+				Verbs:     []string{"create"},
+			},
+		},
+	}
+}
+
+func newEditRole(namespace string) *rbac.ClusterRole {
+	return &rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterRoleName("edit"),
+			Namespace: namespace,
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"persistentvolumeclaims"},
+				Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"persistentvolumeclaims/status"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"cdi.kubevirt.io"},
+				Resources: []string{"datavolumes"},
+				Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+			},
+			{
+				APIGroups: []string{"cdi.kubevirt.io"},
+				Resources: []string{"datavolumes/source"},
+				Verbs:     []string{"create"},
+			},
+		},
+	}
+}
+
+func newViewRoleBinding(namespace string) *rbac.ClusterRoleBinding {
+	return &rbac.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterRoleName("view"),
+			Namespace: namespace,
+		},
+		RoleRef: rbac.RoleRef{
+			Kind:     "Role",
+			Name:     clusterRoleName("view"),
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+		Subjects: []rbac.Subject{{
+			Kind:      "Group",
+			Name:      "system:authenticated",
+			Namespace: namespace,
+			APIGroup:  "rbac.authorization.k8s.io",
+		}},
+	}
+}
+
+func newTemplateBundle() []templatev1.Template {
+	filename := "common-templates-" + version + ".yaml"
+	bundle, err := loadTemplates(filename)
```

Comment:
The template bundle should be cached, so it is not reloaded on every reconciliation iteration.